lib/routes/recovery-keys.js

```diff
@@ -0,0 +1,223 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const errors = require('../error')
+const isA = require('joi')
+const random = require('../crypto/random')
+const requestHelper = require('../routes/utils/request_helper')
+const validators = require('./validators')
+const HEX_STRING = validators.HEX_STRING
+
+module.exports = (log, db, Password, verifierVersion, customs) => {
+  return [
+    {
+      method: 'POST',
+      path: '/recoveryKeys',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            recoveryKeyId: isA.string().max(32).required(),
+            recoveryData: isA.string().max(1024).required()
+          }
+        }
+      },
+      handler(request, reply) {
+        log.begin('createRecoveryKey', request)
+
+        const uid = request.auth.credentials.uid
+        const sessionToken = request.auth.credentials
+        const {recoveryKeyId, recoveryData} = request.payload
+
+        customs.check(request, 'createRecoveryKey')
```

Comment:
This route is session-token authenticated, what do we gain by rate-limiting the creation of these keys?  I think we generally haven't worried too much about rate-limiting for authenticated requests unless it's to prevent accidental client misbehaviour, which seems unlikely here.